src/plugins/nest.js

```diff
@@ -0,0 +1,294 @@
+          }
+
+          let opName = 'nest.pipe.pipesFn'
+          if (pipes.length > 0) {
+            if (pipes[0].constructor && pipes[0].constructor.name) {
+              opName = `${pipes[0].constructor.name}.pipeFn`
+            }
+          }
+
+          const scope = tracer.scope()
+          const childOf = scope.active()
+          const span = tracer.startSpan(opName, {
+            childOf,
+            tags: {
+              'component': 'nest'
+            }
+          })
+
+          return scope.activate(span, () => {
+            try {
+              return pipesFn.apply(this, [args, req, res, next])
+            } catch (e) {
+              throw addError(span, e)
+            } finally {
+              span.finish()
+            }
+          })
+        }
+      }
+      return wrappedPipesFn(createPipesFn.apply(this, arguments))
+    }
+  }
+}
+
+function createWrapCreateGuardsFn (tracer, config) {
+  return function wrapCreateGuardsFn (createGuardsFn) {
+    return function createGuardsFn () {
+      const guards = arguments[0]
+      const instance = arguments[1]
+      let method = null
+      if (arguments.length > 2) {
+        method = arguments[2].name
+      }
+      function wrappedCanActivateFn (canActivateFn) {
+        return (args) => {
+          if (typeof canActivateFn !== 'function') {
+            return canActivateFn
+          }
+
+          let opName = 'nest.guard.canActivate'
+          const scope = tracer.scope()
+          const childOf = scope.active()
+          const span = tracer.startSpan(opName, {
+            childOf,
+            tags: {
+              'component': 'nest'
+            }
+          })
+
+          const guardNames = []
+          guards.forEach(guardName => {
+            guardNames.push(guardName.constructor.name)
+          })
+          if (guardNames.length > 0) {
+            if (guardNames[0].constructor && guardNames[0].constructor.name) {
+              opName = `${guardNames[0]}.canActivate`
+            }
+            span.setOperationName(opName)
+            span.setTag('nest.guards', guardNames)
+          }
+
+          if (instance.constructor && instance.constructor.name) {
+            span.setTag('nest.controller.instance', instance.constructor.name)
+          }
+
+          if (method) {
+            span.setTag('nest.controller.function', method)
+          }
+
+          return scope.activate(span, () => {
+            try {
+              return canActivateFn.apply(this, args)
+            } catch (e) {
+              throw addError(span, e)
+            } finally {
+              span.finish()
+            }
+          })
+        }
+      }
+      return wrappedCanActivateFn(createGuardsFn)
+    }
+  }
+}
+
+function addError (span, error) {
+  span.setTag('error', 'true')
+  span.log({
+    'error.type': error.name,
+    'error.msg': error.message,
+    'error.stack': error.stack
+  })
+  return error
+}
+
+module.exports = [
```

Comment:
nit, can you define the instrumentation in the order the functions are defined (or vice versa)?